design/scheduled-update-check.md

```diff
@@ -0,0 +1,64 @@
+# Scheduled Update Checks for Online Installs
+
+KOTS (Admin Console) does not automatically check for new updates. This proposal adds the 
+ability for KOTS (Admin Console) to check for updates every X period of time for online installations.
+
+## Goals
+
+- Pull new updates automatically for KOTS (Admin Console) online installations
+
+## Non Goals
+
+- Deployment of the new updates
+- Scheduled Update Checks for Airgap Installations
+
+## Background
+
+Many customers assume that scheduled update checks are a given and are surprised when they do
+not receive updates automatically.
+
+## High-Level Design
+
+1. This proposal is only for online installations in KOTS Admin console.
+2. The default period of time (X) is configurable.
+3. The ability to disable this feature completely. It will be enabled by default.
+
+The update check will happen in a background process/thread which is triggered every X period of time.
+On each trigger, the process will get all available applications and run update checks for each app.
+The update check will automatically create new versions for each update available in an app.
+
+## Detailed Design
+
+3 new columns (`update_checker_enabled` & `update_checker_interval` & `update_checker_status`) will be 
+added to the `app` table in the postgress database using a migration.
+
+* The `update_checker_enabled` column is a bool and defaults to `true`. 
+* The `update_checker_interval` column is a text and defaults to `6h`.
+* The `update_checker_status` column is a text and defaults to `NULL`
```

Comment:
There is already a mechanism for tracking update status.  Scheduled update checks should also be visible on the UI and the user should not be able to click the `Check for updates` button while a scheduled check is running.  This is already built, so let's use it.

Reply:
This is the status of the checker, not the update itself. To check if it is running or not, and why.

Reply:
I don't think that's needed.  There is no way to view or use this info.  And you can only run 1 update check at a time, manual and scheduled.

Reply:
There is a way to view this info, it is mentioned here https://github.com/replicatedhq/kots/pull/497/files#diff-ea548131f69005447776a7f865c3c0b2R54

Reply:
How would a customer know if automatic update checks are even running then? the value in the database can indicate it's enabled, but an error can occur and the customer will never know.

Reply:
That's already built.  If you click "check for updates" and there is an error, it will be shown on the UI.  There should be no difference in manual and scheduled update checks other than how they are triggered.

Reply:
I don't get how that's already built. There is a new thing introduced here "Update Checker". If the checker is not running, there will be _no update_ to fail with an error in the first place.